core/src/test/java/org/elasticsearch/cluster/routing/allocation/MaxRetryAllocationDeciderTests.java

```diff
@@ -208,4 +213,45 @@ public void testFailedAllocation() {
         assertEquals(routingTable.index("idx").shard(0).shards().get(0).state(), INITIALIZING);
         assertEquals(routingTable.index("idx").shard(0).shards().get(0).unassignedInfo().getMessage(), "ZOOOMG");
     }
+
+    /**
+     * Tests that {@link MaxRetryAllocationDecider#canForceAllocatePrimary(ShardRouting, RoutingNode, RoutingAllocation)} adheres
+     * to the maximum retry count and returns a NO decision if the maximum number of retries is exceeded.
+     */
+    public void testCanForceAllocatePrimary() {
```

Comment:
I removed this test and put the canForceAllocatePrimary checks in `testFailedAllocation`